### ISSUE TITLE
Enable date ranges per layer to populate Year facet and range plugin

### ISSUE THREAD

**Issue by karenmajewicz** [geoblacklight/geoblacklight#746]
### Descriptive summary

Many layers represent resources that span a time period. However, the Year facet and range plugins are typically set to be populated by a single integer in solr_year_i.  There has not been consensus on what year to choose when there is a range- some choose start, some choose end date.

If the facets can be changed to accept multiple years, a multivalued metadata field could be filled with multiple single years, such as how Leventhal has it configured here: https://collections.leventhalmap.org/search/commonwealth:z603vn07z.json

However, simply adding a range (such as 1800-1825) would be an easier value to create in a metadata record.

**Comment by andrewbattista** [comment]
To me this seems as much of a metadata question as it is a technical one. I think the only reason we have the `solr_year_i` value is to facilitate the use of the [date range limit plug-in](https://github.com/projectblacklight/blacklight_range_limit), right? Are you proposing work on the tools to author metadata or just the functionality of this plug-in being extended somehow?

**Comment by mejackreed** [comment]
Yes this is to out of the box support the date range plugin. Which can handle an array of years

`[1900, 1901, 1902, ... ]`

But from a metadata perspective I realize that is less desirable. 

We could do something better int he metadata, but it would require custom indexing, and would remove some of the functionality of direct posting of solr records to solr.

**Comment by andrewbattista** [comment]
Right... it seems like it would be a bit easier to generate values in metadata or remediate them systematically, but what we really want is the function of the slider/histogram with the ability to recognize records that have many discrete years in them.

**Comment by thatbudakguy** [comment]
I think Aardvark's [index year](https://opengeometadata.org/ogm-aardvark/#index-year) solves this problem pretty neatly. We're generating it at index time and using it to power the hierarchical data search in current EarthWorks.

Transforming a different field like "date range" into a sequence of numeric years seems like an index-time/metadata problem rather than one for the application to handle, so I'm going to consider this done.
